packages/flutter/lib/src/material/menu_bar_theme.dart

```diff
@@ -0,0 +1,333 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'menu_bar.dart';
+import 'theme.dart';
+
+/// Defines the visual properties of [MenuBar], [MenuBarMenu] and
+/// [MenuBarButton] widgets.
+///
+/// Descendant widgets obtain the current [MenuBarThemeData] object
+/// using `MenuBarTheme.of(context)`. Instances of
+/// [MenuBarThemeData] can be customized with
+/// [MenuBarThemeData.copyWith].
+///
+/// Typically, a [MenuBarThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.menuBarTheme]. Otherwise,
+/// [MenuBarTheme] can be used to configure its own widget subtree.
+///
+/// All [MenuBarThemeData] properties are `null` by default.
+/// If any of these properties are null, the menu bar will provide its own
+/// defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class MenuBarThemeData with Diagnosticable {
+  /// Creates the set of properties used to configure [MenuBarTheme].
+  const MenuBarThemeData({
+    this.barHeight,
+    this.barPadding,
+    this.barBackgroundColor,
+    this.barElevation,
+    this.menuBackgroundColor,
+    this.menuElevation,
+    this.menuShape,
+    this.menuPadding,
+    this.itemBackgroundColor,
+    this.itemForegroundColor,
+    this.itemOverlayColor,
+    this.itemTextStyle,
+    this.itemPadding,
+    this.itemShape,
+  });
+
+  /// The minimum height of the menu bar.
+  final double? barHeight;
+
+  /// The padding around the outside of a [MenuBar].
+  final EdgeInsets? barPadding;
+
+  /// The background color of the [MenuBar].
+  final MaterialStateProperty<Color?>? barBackgroundColor;
+
+  /// The Material elevation of the [MenuBar].
+  ///
+  /// See also:
+  ///
+  ///  * [Material.elevation] for a description of how elevation works.
+  final MaterialStateProperty<double?>? barElevation;
+
+  /// The background color of a [MenuBarMenu].
+  final MaterialStateProperty<Color?>? menuBackgroundColor;
+
+  /// The Material elevation of the [MenuBarMenu].
+  ///
+  /// See also:
+  ///
+  ///  * [Material.elevation] for a description of how elevation works.
+  final MaterialStateProperty<double?>? menuElevation;
+
+  /// The shape around a [MenuBarMenu].
+  final MaterialStateProperty<ShapeBorder?>? menuShape;
+
+  /// The padding around the outside of a [MenuBarMenu].
```

Comment:
I haven't "read ahead" but as with menu bar padding, I'm wondering why this property is needed.